Training25/Program.cs

```diff
@@ -3,11 +3,50 @@
 // Copyright (c) Metamation India.
 // ------------------------------------------------------------------
 // Program.cs
-// Program on main branch.
+// Program that prints sorted array by keeping the elements matching special character S
+// to the last of the array
 // ------------------------------------------------------------------------------------------------
+using System.Text.RegularExpressions;
+using static System.Console;
+
 namespace Training25;
+
 internal class Program {
-   static void Main (string[] args) {
-      Console.WriteLine ("Hello, World!");
+
+   static void Main () {
+      const string msg = """
+               Ensure input is of the following format:
+               ([a, b, c, a, c, b, d], a, "descending")
+                         A             S        O
+               A - Character array - string of letters separated by comma(',') and space(' ')
+               S - Special character
+               O - Sort order (optional)
+               """;
+      do {
+         Clear ();
+         Write ($"{msg}\nEnter string: ");
+         var input = ReadLine () ?? "";
+         if (!string.IsNullOrEmpty (input) &&
+            Regex.IsMatch (input, @"\(\[(.*?)\], \w(, ""(ascending|descending)"")?\)")) {
+            input = input[2..^1];
+            var str = input.Replace (", ", "").Split (']');
+            var arr = str[0];
+            if (arr.All (char.IsLetter)) {
+               char spl = str[1][0]; string order = str[1][1..];
+               WriteLine (SortAndSwap (arr, spl, order == "\"descending\""));
+            } else WriteLine ("Invalid format! Please refer format for character array(A)!");
+         } else WriteLine ("Invalid Format! Please try again!");
+         Write ("Press 'Y' to continue");
+      } while (ReadKey (true).Key == ConsoleKey.Y);
+   }
+
+   // Returns the sorted array of letters.
+   // Letters that match the special character are added at the end, if any.
+   static string SortAndSwap (string str, char splChar, bool isDescending) {
+      var arr = str.Where (a => a != splChar);

```

Comment:
```suggestion
      var arr = str.Where (a => a != splChar).ToList ();
```
Can convert to list here as you are calling Count () once and again using arr in string.Join which means enumeration will be done twice.